### PR TITLE
Added allowNsfw option

### PR DESCRIPTION
**Pull request by qtdceu**
Added the allowNsfw feature. Default is false, when true it will not filter out results containing NSFW. E.g.

```js
// No NSFW 
RedditImageFetcher.fetch({
   type: 'wallpaper',
   total: 50,
});

// Allowing NSFW
RedditImageFetcher.fetch({
   type: 'wallpaper',
   total: 50,
   allowNsfw: true
});
```

### CODE REVIEW

**Comment by arifszn** [comment]
Thanks. Will review it soon today.

File: index.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const fetch = async (options = {}) => {
         let total = 1;
         let type = 'meme';
         let subreddit = config.memeSubreddit;
+        let allowNsfw = false;
```
Comment (arifszn) [inline index.js:19]: Let's keep the default as true.

**Review by arifszn — CHANGES_REQUESTED** [review]
Awesome work. Please check the comments.

File: README.md [hunk 2]
```diff
@@ -150,6 +150,7 @@ RedditImageFetcher.fetch({
 | subreddit        | [string]   | Custom subreddit library                    |   <code>[ ]</code>   |
 | addSubreddit        | [string]   | Add subreddits to subreddit library                    | <code>[ ]</code>     |
 | removeSubreddit     | [string]   | Remove subreddits from subreddit library               | <code>[ ]</code>     |
+| allowNsfw     | boolean   | Allow NSFW results              | <code>false</code>     |
```
Comment (arifszn) [inline README.md:153]: Change this to `allowNSFW`.

File: index.js [hunk 3]
```diff
@@ -38,6 +39,12 @@ const fetch = async (options = {}) => {
                 }
             }
 
+            if (typeof options.allowNsfw !== 'undefined') {
```
Comment (arifszn) [inline index.js:42]: Add the type definition in `index.d.ts`

**Review by arifszn — CHANGES_REQUESTED** [review]
Awesome work. Please check the comments.

**Comment by qtdceu** [comment]
All changes requested have been implemented.

File: index.js [hunk 4]
```diff
@@ -38,6 +39,12 @@ const fetch = async (options = {}) => {
...
+                if (options.allowNsfw === true) {
+                    allowNsfw = true;
+                };
```
Comment (arifszn) [inline index.js:45]: Reverse this logic?

Comment (qtdceu) [reply index.js]: My bad, should be fixed now.

**Review by arifszn — APPROVED** [review]

**Pull request merged**